x/gov/keeper.go

```diff
@@ -88,32 +94,35 @@ func NewKeeper(cdc *codec.Codec, key sdk.StoreKey, paramsKeeper params.Keeper,
 		paramSpace:   paramSpace.WithKeyTable(ParamKeyTable()),
 		ck:           ck,
 		ds:           ds,
+		router:       NewRouter(),
 		vs:           ds.GetValidatorSet(),
 		cdc:          cdc,
 		codespace:    codespace,
 	}
 }
 
+// TODO: implement seal
+// Returns router
+func (keeper Keeper) Router() Router {
```

Comment:
Can this be private?

Reply:
The root application need to have router in order to register handlers. We can however make `NewKeeper` returns `(Keeper, Router)` so only the logic who called the constructor can access on the router.